actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -339,7 +339,10 @@ def method_missing(called, *args, **options, &block)
 
             TagHelper.ensure_valid_html5_tag_name(name)
 
-            tag_string(name, *args, **options, &block)
+            escape = options.fetch(:escape, true)
+            options = options.except(:escape)
```

Comment:
Does moving `escape: true` to the method signature net any gains?

```ruby
def method_missing(name, *args, escape: true, **options, &block)
  name = called.to_s.dasherize

  TagHelper.ensure_valid_html5_tag_name(name)

  tag_string(name, *args, escape: escape, **options, &block)
end
```

Reply:
Surprisingly, I'm not seeing much difference in performance between the two (although I agree your suggestion looks nicer so I'll apply it anyways). Either way, `method_missing` is about 50% slower than tag_builder